### PR TITLE
Adding AAVEv2 flashloan provider support and tests

### PR DESCRIPTION
**Pull request by iphelix**
Sample usage of the new provider here:

https://github.com/iphelix/euler-exploit-poc

Fixes: N/A

## Vulnerability Type
N/A

## Usage
Sample usage is implemented in the FlashLoanExample.t.sol and can be executed as follows:

```
forge test --match-path test/examples/FlashLoanExample.t.sol -vv
```

#### PR Checklist

- [ x ] Attack Template
- [ x ] Attack Example
- [ x ] Test
- [ ] Readme Updated

### CODE REVIEW

File: src/flashloan/FlashLoanProvider.sol [hunk 1]
```diff
@@ -31,6 +33,8 @@ library FlashLoanProvider {
             BalancerFlashLoan.takeFlashLoan(tokens, amounts);
         } else if (flp == FlashLoanProviders.UNISWAPV2) {
             UniswapV2FlashLoan.takeFlashLoan(tokens, amounts);
+        } else if (flp == FlashLoanProviders.AAVEV2) {
+            AAVEV3FlashLoan.takeFlashLoan(tokens, amounts);
```
Comment (janbro) [inline src/flashloan/FlashLoanProvider.sol:37]: Looks like this should be calling AAVEV2FlashLoan lib

**Review by janbro — APPROVED** [review]
Approving to run GH workflows

File: src/flashloan/FlashLoanProvider.sol (hunk 1 above)
Comment (iphelix) [reply src/flashloan/FlashLoanProvider.sol]: Oops, updated.

File: src/flashloan/examples/FlashLoanExample.sol [hunk 2]
```diff
@@ -11,7 +11,7 @@ contract FlashLoanExample is FlashLoan, Tokens {
     function initiateAttack(FlashLoanProviders flp) external {
         _flp = flp;
         uint256 fee;
-        if (flp == FlashLoanProviders.AAVEV1 || flp == FlashLoanProviders.AAVEV3) {
+        if (flp == FlashLoanProviders.AAVEV1 || flp == FlashLoanProviders.AAVEV2 ||flp == FlashLoanProviders.AAVEV3) {
```
[the remark this answers is not in the document]
Comment (iphelix) [reply src/flashloan/examples/FlashLoanExample.sol]: Removed space and ran `forge fmt`.

**Comment by janbro** [comment]
Really appreciate the contribution! Thank you for keeping the logic consistent, this is great work 🎉  Will add the provider and supported chains in a new PR which will be fixing GH workflows

**Pull request merged**